feature/experimental/p4rt/otg_tests/traceroute_packetin_test/README.md

```diff
@@ -1,34 +1,125 @@
 # P4RT-5.1: Traceroute: PacketIn
 
-
 ## Summary
 
-Verify that Traceroute packets are punted with correct metadata.
+Test FRR behaviors with encapsulation scenarios.
+
+## Topology
+
+ATE port-1 <------> port-1 DUT
+DUT port-2 <------> port-2 ATE
+DUT port-3 <------> port-3 ATE
+DUT port-4 <------> port-4 ATE
+DUT port-5 <------> port-5 ATE
+DUT port-6 <------> port-6 ATE
+DUT port-7 <------> port-7 ATE
+DUT port-8 <------> port-8 ATE
+
+## Baseline setup
+
+*   Setup equivalent to [TE-17.1 vrf_policy_driven_te](https://github.com/openconfig/featureprofiles/blob/main/feature/experimental/gribi/otg_tests/vrf_policy_driven_te/README.md), including GRibi programming.
+
+*   Install a BGP route resolved by ISIS in default VRF to route traffic out of DUT port-8 for 1.2.3.4.
+
+*   Enable the P4RT server on the device.
+
+*   Connect a P4RT client and configure the forwarding pipeline. Install P4RT table entries required for traceroute.
+	These are located in [p4rt_utils.go] (https://github.com/openconfig/featureprofiles/blob/main/internal/p4rtutils/p4rtutils.go)
+	p4rtutils.ACLWbbIngressTableEntryGet(packetIO.GetTableEntry(delete, isIPv4))
 
 
 ## Procedure
 
-*	Connect ATE port-1 to DUT port-1, ATE port-2 to DUT port-2.
-*	TODO: Install the set of routes on the device.
-*	Enable the P4RT server on the device.
-*	Connect a P4RT client and configure the forwarding pipeline. InstallP4RT table 	entries required for traceroute.
-*	Send IPv4 packets from the ATE with TTL=1 and verify that packets with TTL=1 are received by the client.
-*	Send IPv6 packets from the ATE with HopLimit=1 and verify that packets with HopLimit=1 are received by the client.
-*	Verify that the packets have both ingress_singleton_port and egress_singleton_port metadata set.
+At the start of each of the following scenarios, ensure:
+
+*   All ports are up and baseline is reset as above.
+
+Unless otherwise specified, all the tests below should use traffic with
+`dscp_encap_a_1` referenced int he VRF selection policy.
+
+### Test-1
+
+Tests that traceroute with TTL=1 matched the VRF selection policy for encap.
+
+*   Send packets to DUT port-1 with TTL=1. The outer v4 header has the destination
+    addresses 138.0.11.8. verify that packets with TTL=1 are received by the client.
+
+*   Verify that the packets have both ingress_singleton_port and egress_singleton_port metadata set.
+egress_singletone_port should have port 2 with weight 1, port 3 weight 3, and port 4 with weight 6.
```

Comment:
is the expectation that the metadata returns RIB/FIB table results instead of a result of the FIB lookup? This appears to be a new requirement / a new use of this metadata field. Is the format of this type of response detailed somewhere?

Reply:
The expectation I'm trying to encode is that traceroute packets respect the VRF selection policy - I'll update to make this more clear